extensions/wikia/Scribe/ScribeEventProducer.class.php

```diff
@@ -41,35 +42,63 @@ function __construct( $key, $archive = 0 ) {
 
 		$this->setCityId( $this->app->wg->CityId );
 		$this->setServerName( $this->app->wg->Server );
-		$this->setIp( $this->app->wg->Request->getIP() );
+		$this->setIP( $this->app->wg->Request->getIP() );
 		$this->setHostname( wfHostname() );
 		$this->setBeaconId ( wfGetBeaconId() );
 		$this->setArchive( $archive );
 		$this->setLanguage();
 		$this->setCategory();
+		$this->setErrorTemplateMessage();
 	}
 
-	public function buildEditPackage( $oPage, $oUser, $oRevision = null, $oLocalFile = null ) {
-		wfProfileIn( __METHOD__ );
+	/**
+	 * @param $oUser User
+	 * @return bool
+	 */
+	public function isUser( $oUser ) {
+		if ( !$oUser instanceof User ) {
+			\Wikia\Logger\WikiaLogger::instance()->error( $this->errorTemplateMsg . "invalid user object" );
+			return false;
+		}
+		return true;
+	}
 
+	/**
+	 * @param $oPage WikiPage
+	 * @return bool
+	 */
+	public function isObject( $oPage ) {
 		if ( !is_object( $oPage ) ) {
-			Wikia::log( __METHOD__, "error", "Cannot send log using scribe ({$this->app->wg->CityId}): invalid WikiPage object" );
-			wfProfileOut( __METHOD__ );
+			\Wikia\Logger\WikiaLogger::instance()->error( $this->errorTemplateMsg . "invalid WikiPage object" );
 			return false;
 		}
+		return true;
+	}
 
-		if ( !$oUser instanceof User ) {
-			Wikia::log( __METHOD__, "error", "Cannot send log using scribe ({$this->app->wg->CityId}): invalid user object" );
-			wfProfileOut( __METHOD__ );
+	/**
+	 * @param $oRevision Revision
+	 * @return bool
+	 */
+	public function isRevision( $oRevision ) {
+		if ( !empty( $oRevision ) && !$oRevision instanceof Revision ) {
+			\Wikia\Logger\WikiaLogger::instance()->error( $this->errorTemplateMsg . "invalid revision object" );
 			return false;
 		}
+		return true;
+	}
 
-		if ( !empty( $oRevision ) ) {
-			if ( !$oRevision instanceof Revision ) {
-				Wikia::log( __METHOD__, "error", "Cannot send log using scribe ({$this->app->wg->CityId}): invalid revision object" );
-				wfProfileOut( __METHOD__ );
-				return false;
-			}
+	/**
+	 * @param $oPage WikiPage
+	 * @param $oUser User
+	 * @param null $oRevision Revision
+	 * @return bool
+	 */
+	public function buildEditPackage( $oPage, $oUser, $oRevision = null ) {
+		wfProfileIn( __METHOD__ );
+
+		if ( !( $this->isObject( $oPage ) && $this->isUser( $oUser ) && $this->isRevision( $oRevision ) ) ) {
+			wfProfileOut( __METHOD__ );
+			return false;
 		}
 
 		$revision_id = $page_id = $page_namespace = 0;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "$page_namespace" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)